In [1]:
import numpy as np
import pandas as pd

from statsmodels.stats.weightstats import ttest_ind
from statsmodels.stats.proportion import proportions_ztest

In [2]:
def get_hits(hit_column):
    """Get the total number of hits / total number of successful hits
    for both weapons
    
    Args:
        hit_column (str): 
            column containing the hit data 
            'total_successful_hits' or 'total_hits'
        
    Returns:
        num_hits (array): 
            total number of (successful) hits for the Event RPG and
            Iron Scimitar       
    """
    
    num_hits = exp_data[hit_column].to_numpy().reshape(2, -1).max(axis=1)
    return num_hits

In [3]:
exp_data = pd.read_csv('data/processed_experience_data.csv')

In [4]:
num_successes = get_hits('total_successful_hits')
num_observations = get_hits('total_hits')

In [5]:
ztest_results = proportions_ztest(num_successes, num_observations)
p_success = (num_successes / num_observations)
print('p-value:', ztest_results[1])
print('\nEstimate for p1:', p_success[0].round(3))
print('Estimate for p2:', p_success[1].round(3))

p-value: 0.0017501372654863242

Estimate for p1: 0.19
Estimate for p2: 0.215


Letting $p_{1}$ and $p_{2}$ represent the probability of landing a successful hit when training with the Event RPG and Iron Scimitar respectively, the Z-test tests the following hypotheses: <br>
$H_{0}$: $p_{1}$ = $p_{2}$  
$H_{1}$: $p_{1}$ ≠ $p_{2}$ <br>

Given that the p-value is less than 0.05, we can reject $H_{0}$ in favour of $H_{1}$ which suggests that the probabilities of landing a successful hit when using the Iron Scimitar and Event RPG are different.

On a side note, the sample proportion (number of successful hits divided by the total number of hits) can be used as unbiased estimators for probabilities $p_{1}$ and $p_{2}$.

$\hat{p}_{1}$ = 0.190 <br>
$\hat{p}_{2}$ = 0.215

In [6]:
exp = exp_data['exp_gained'].to_numpy().reshape(2, -1)
rpg_exp, scim_exp = exp[0, ], exp[1, ]

In [7]:
ttest_results = ttest_ind(rpg_exp, scim_exp)
print('p-value:', ttest_results[1])

p-value: 0.06001573332108156


Letting $u_{1}$ and $u_{2}$ represent the mean experience per 10 minutes when training with the Event RPG and Iron Scimitar respectively, the T-test tests the following hypotheses: <br>
$H_{0}$: $u_{1}$ = $u_{2}$  
$H_{1}$: $u_{1}$ ≠ $u_{2}$ <br>

Since the p-value is greater than 0.05, we cannot reject $H_{0}$.